Modules/Sources/AsyncImageKit/ImageDownloader.swift

```diff
@@ -46,6 +47,20 @@ public final class ImageDownloader {
     }
 
     public func data(for request: ImageRequest) async throws -> Data {
+
+        if case .video(let url) = request.source {
+            let asset = AVURLAsset(url: url)
+            let generator = AVAssetImageGenerator(asset: asset)
+            let result = try await generator.image(at: CMTime(seconds: 0.0, preferredTimescale: 600))
+            let image = UIImage(cgImage: result.image)
+
+            guard let data = image.pngData() else {
```

Comment:
It's relatively expensive to convert it to data and back. Does it needs to support `data(for request)`. Would make sense to move it under `image(for request)`? Alternately, `data(for request)` could call `image(for request)` and then convert it to data so both would be supported.

Reply:
This is addressed in #25000